code/game/machinery/computer/sm_monitor.dm

```diff
@@ -0,0 +1,142 @@
+/obj/machinery/computer/sm_monitor
+	name = "supermatter monitoring console"
+	desc = "Used to monitor supermatter shards."
+	icon_keyboard = "power_key"
+	icon_screen = "smmon_0"
+	circuit = /obj/item/circuitboard/sm_monitor
+	light_color = LIGHT_COLOR_YELLOW
+	/// Cache-list of all supermatter shards
+	var/list/supermatters = list()
+	/// Last status of the active supermatter for caching purposes
+	var/last_status
+	/// Reference to the active shard
+	var/obj/machinery/power/supermatter_shard/active
+
+/obj/machinery/computer/sm_monitor/Destroy()
+	active = null
+	return ..()
+
+/obj/machinery/computer/sm_monitor/attack_ai(mob/user)
+	attack_hand(user)
+
+/obj/machinery/computer/sm_monitor/attack_hand(mob/user)
+	add_fingerprint(user)
+	if(stat & (BROKEN|NOPOWER))
+		return
+	tgui_interact(user)
+
+/obj/machinery/computer/sm_monitor/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "SupermatterMonitor", name, 600, 325, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/sm_monitor/tgui_data(mob/user)
+	var/list/data = list()
+
+	if(istype(active))
+		var/turf/T = get_turf(active)
+		// If we somehow delam during this proc, handle it somewhat
+		if(!T)
+			active = null
+			refresh()
+			return
+		var/datum/gas_mixture/air = T.return_air()
+		if(!air)
+			active = null
+			return
+
+		data["active"] = TRUE
+		data["SM_integrity"] = active.get_integrity()
+		data["SM_power"] = active.power
+		data["SM_ambienttemp"] = air.temperature
+		data["SM_ambientpressure"] = air.return_pressure()
+		//data["SM_EPR"] = round((air.total_moles / air.group_multiplier) / 23.1, 0.01)
+		var/other_moles = air.total_trace_moles()
+		var/TM = air.total_moles()
+		if(TM)
+			data["SM_gas_O2"] = round(100*air.oxygen/TM,0.01)
+			data["SM_gas_CO2"] = round(100*air.carbon_dioxide/TM,0.01)
+			data["SM_gas_N2"] = round(100*air.nitrogen/TM,0.01)
+			data["SM_gas_PL"] = round(100*air.toxins/TM,0.01)
```

Comment:
[Spacing between the operators and comma](https://github.com/ParadiseSS13/Paradise/blob/master/.github/CONTRIBUTING.md#spacing).

Reply:
This was copied from the old version but ill fix this one's anyway